app/components/works/show/related_contents_show_component.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Works
+  module Show
+    # Component for rendering related contents on the work show page
+    class RelatedContentsShowComponent < ApplicationComponent
+      def initialize(work_presenter:)
+        @work_presenter = work_presenter
+        super()
+      end
+
+      attr_reader :work_presenter
+
+      delegate :related_links, to: :work_presenter
+
+      def related_works
+        work_presenter.related_works_attributes.select { |related_work| related_work.to_s.present? }
+      end
+
+      def relationship_label_for(related_work)
+        t("related_works.edit.fields.relationship.options.#{related_work.relationship}")
```

Comment:
Will all values of `related_work.relationship` result in a translation being found in our locales? If not, the return value will be a missing translation string. If that's an issue, we can consider:

```suggestion
        t("related_works.edit.fields.relationship.options.#{related_work.relationship}", default: nil)
```

Which will get us a `nil` relationship label.

Reply:
They all have translations.